inst/string.m

```diff
@@ -148,23 +149,30 @@
           error ('string: cell inputs must be cellstr');
         endif
         this.strs = in;
-        this.tfMissing = false (size (this.strs));
+        tfM = false (size (this.strs));
+        tfM(cellfun (@(x) isempty(x), this.strs)) = true;
+        this.tfMissing = tfM;
       elseif isnumeric (in)
         this.strs = arrayfun (@(x) {num2str(x)}, in);
-        this.tfMissing = false (size (this.strs));
+        tfM = false (size (this.strs));
+        tfM(arrayfun (@(x) isnan(x), in)) = true;
+        this.tfMissing = tfM;
       elseif isa (in, 'datetime')
         this.strs = arrayfun (@(x) {datestr(x)}, in);
         this.tfMissing = false (size (this.strs));
       elseif isa (in, 'duration') || isa (in, 'calendarDuration')
-        error ('string: duration and calendarDuration conversion are not implemented yet. Sorry.');
+        this.strs = arrayfun (@(x) dispstrs(x), in, "UniformOutput", false){:};
+        tfM = false (size (this.strs));
+        tfM(arrayfun (@(x) isnan(x), in, "UniformOutput", false){:}) = true;
```

Comment:
`isnat` would be more "correct" than `isnan` here. `isnat` is the standard one to use with date/time types; the duration classes' support for `isnan` is a compatibility hack I threw in mostly because I wanted to use `isnan` polymorphically in a generic context, but here the input is known to support `isnat` and genericity isn't needed. And now that `ismissing` is a well-supported thing (at least on the Matlab side), `ismissing` replaces the need for a generification of isnan.

The `arrayfun` may be unnecessary, because `isnat` on duration types is array-oriented. Could maybe be reduced to `this.tfMissing = isnat(in);`?